Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/service/impl/ConceptMergeService.java

```diff
@@ -0,0 +1,224 @@
+import edu.asu.conceptpower.app.service.IConceptMergeService;
+import edu.asu.conceptpower.core.ConceptEntry;
+
+@Service
+public class ConceptMergeService implements IConceptMergeService {
+
+    @Autowired
+    private IConceptManager conceptManager;
+
+    @Autowired
+    private ConceptTypesService conceptTypesService;
+
+    @Override
+    public ConceptsMergeBean prepareMergeConcepts(List<ConceptEntry> conceptEntries,
+            ConceptsMergeBean conceptsMergeBean) {
+        boolean typeWarning = false;
+        boolean posWarning = false;
+        conceptsMergeBean.clear();
+        for (ConceptEntry entry : conceptEntries) {
+            if (entry.getWord() != null && !entry.getWord().isEmpty()) {
+                conceptsMergeBean.setWord(conceptsMergeBean.getWord() == null
+                        || conceptsMergeBean.getWord().length() < entry.getWord().length() ? entry.getWord()
+                                : conceptsMergeBean.getWord());
+                
+                String conceptIdWithName = entry.getWord().trim() + " -- " + entry.getId();
+                conceptsMergeBean.getConceptNamesWithIds().add(conceptIdWithName);
+            }
+
+            if (conceptsMergeBean.getSelectedPosValue() == null) {
+                conceptsMergeBean.setSelectedPosValue(entry.getPos().toLowerCase());
+            } else if (!conceptsMergeBean.getSelectedPosValue().equalsIgnoreCase(entry.getPos())) {
+                posWarning = true;
+            }
+
+            if (entry.getDescription() != null && !entry.getDescription().isEmpty()) {
+                conceptsMergeBean.getDescriptions().add(entry.getDescription().trim());
+            }
+
+            if (entry.getSynonymIds() != null && !entry.getSynonymIds().isEmpty()) {
+                conceptsMergeBean.getSynonymsids().addAll(Arrays.asList(entry.getSynonymIds().split(",")));
+            }
+
+            if (entry.getEqualTo() != null && !entry.getEqualTo().isEmpty()) {
+                conceptsMergeBean.getEqualsValues().add(entry.getEqualTo());
+            }
+            if (entry.getSimilarTo() != null && !entry.getSimilarTo().isEmpty()) {
+                conceptsMergeBean.getSimilarValues().add(entry.getSimilarTo());
+            }
+
+            if (entry.getTypeId() != null && conceptsMergeBean.getSelectedTypeId() == null) {
+                conceptsMergeBean.setSelectedTypeId(entry.getTypeId());
+            } else if (entry.getTypeId() != null) {
+                typeWarning = true;
+            }
+
+            if (entry.getWordnetId() != null && !entry.getWordnetId().isEmpty()) {
+                String[] wordNetIds = entry.getWordnetId().split(",");
+                conceptsMergeBean.getMergedIds()
+                        .addAll(Arrays.asList(wordNetIds).stream()
+                                .filter(wordNetId -> !wordNetId.trim().equalsIgnoreCase("")).map(wordNetId -> wordNetId)
+                                .collect(Collectors.toSet()));
```

Comment:
Why are wordnet ids added to the mergedId list here, but the actually merged concepts are added later? It doesn't make sense and takes two things apart that belong together. Is there a reason?